cpp/src/parquet/encoding.cc

```diff
@@ -480,11 +480,18 @@ class DictEncoderImpl : public EncoderImpl, virtual public DictEncoder<DType> {
  public:
   typedef typename DType::c_type T;
 
-  explicit DictEncoderImpl(const ColumnDescriptor* desc, MemoryPool* pool)
-      : EncoderImpl(desc, Encoding::PLAIN_DICTIONARY, pool),
+  explicit DictEncoderImpl(const ColumnDescriptor* desc, Encoding::type dict_encoding,
```

Comment:
To provide some detail, the encoding type of dictionary encoded pages is depending on the format version
- Format v1: `PLAIN_DICTIONARY` is used for both dict page and data page. 
- Format v2: dict page uses `PLAIN` and data page uses `RLE_DICTIONARY`

So it is kind of weird to manually pass `dict_encoding` because they are the same thing but relevant to different parquet formats.

Reply:
Right, currently they will be decided by `dictionary_index_page()` in `WriterProperties`

And currently, there encoding logic are similiar